wiki/osu!_File_Formats/Osu_(file_format)/zh.md

```diff
@@ -0,0 +1,365 @@
+
+## Colours（颜色）
+
+本节中的所有选项均代表颜色。它们是逗号分隔的三元组，整数为0-255，代表红色，绿色和蓝色的百分比占量（RGB）。
+
+| 选项 | 描述 |
+| :-- | :-- |
+| `Combo#`，`#` 为整数 | 附加的连击颜色 |
+| `SliderTrackOverride` | 附加的滑条轨道颜色 |
+| `SliderBorder` | 滑块边框颜色 |
+
+## Hit objects（物件）
+
+*物件语法:* `x,y,time,type,hitSound,objectParams,hitSample`
+
+- **`x` （整数）** 和 **`y` （整数）：** 以 [osu! 像素](/wiki/Glossary#osupixel)为单位的物件位置。
+- **`time` （整数）：** 物件应该被击中的时间，从谱面音频开始以毫秒为单位的时间。
+- **`type` （整数）：** 用位符表示的物件类型。 参见 [物件类型小节](#Type（物件类型）).
+- **`hitSound` （整数）：** 用位符表示物件所应用的打击音效，参见 [打击音效小节](#Hitsounds（打击音效）).
+- **`objectParams` （由逗号分隔的列表）：** 特定于物件类型的额外参数。
+- **`hitSample` （由逗号分隔的列表）：** 有关在击中对象时播放哪些样本的信息。 它与`hitSound`密切相关； 参见 [打击音效小节](#Hitsounds（打击音效）) 。如果未被填写，默认为 `0:0:0:0:`。
+
+### Type（物件类型）
+
+命中对象类型以8位整数存储，其中每个位都具有特殊的含义。基础的物件类型由0，1，3， 和7 （从最低到最高）组成：
+
+- 0：打击圈
+- 1：滑条
+- 3：转盘
+- 7：osu!mania滑条
+
+剩余的位用于区分新的连击并有选择地跳过连击颜色：
+
+- 2：新连击
+- 4–6: 一个3位整数，指定此物件成为新的连击时要跳过的连击颜色数。
+
+### Hitsounds（打击音效）
+
+`hitSound` 位标记决定物件被击中时播放的音效：
+
+- 0：Normal
+- 1：Whistle
+- 2：Finish
+- 3: Clap
+
+如果没有设置，Normal将用作默认音效。
+
+除了osu!mania模式， `LayeredHitSounds` 属性强制将Normal包括在内，而与bit 0的设置无关。 默认开启。
+
+#### 自定义采样集
+
+使用`hitSample`可以进一步定义播放的音效，如果未填写默认为`0:0:0:0:` 。
+
+*Hit sample 语法：* `normalSet:additionSet:index:volume:filename`
+
+- **`normalSet` （整数）：** 普通音效的采样集。
+- **`additionSet` （整数）：** whistle，finish和clap的采样集。
+- **`index` （整数）：** 采样集索引，如果设置为`0`，将使用时间点所设置的采样集索引。
+- **`volume` （整数）：** 采样集的音量（1到100）。如果设置为`0`，将使用时间点所设置的音量。
+- **`filename` （字符串）：** 自定义外部音效文件名。
+
+`normalSet` 和 `additionSet` 可以是下面的任何一个：
+
+- `0`：无自定义采样集
+  - 对于普通打击音效，采样集由时间点采样集决定。
+  - 对于附加打击音效， 采样集由普通打击音效采样集决定。
+- `1`：Normal 集合
+- `2`：Soft 集合
+- `3`：Drum 集合
+
+所有这些选项（音量除外）都用于确定给定的匹配声音播放哪个声音文件。文件名为 `<sampleSet>-hit<hitSound><index>.wav`，其中：
+
+- `sampleSet` 为 `normal`，`soft`或者 `drum`，由`normalSet` 或者 `additionSet` 其中任意一个决定播放的音效
+- `hitSound` 为 `normal` ，`whistle`，`finish`，或者`clap`
+- `index` 与上面的 `index` 一样，如果是`0`和`1`则可以忽略。
+
+从以下第一个含匹配文件名的目录中加载声音文件：
+
+- 谱面文件夹，如果`index`不是`0`
+- 皮肤，如果`index`被移除
+- osu!默认资源，如果`index`被移除
+
+当填写了`filename`，不会播放附加声音，取而代之播放谱面文件夹的这个文件
+
+### Hit circles（打击圈）
+
+打击圈没有附加的`objectParams`.
+
+### Sliders（滑条）
+
+*滑条语法：* `x,y,time,type,hitSound,curveType|curvePoints,slides,length,edgeSounds,edgeSets,hitSample`
+
+- **`curveType` （字母）：**此滑块的曲线类型 (`B` = 贝兹曲线, `C` = 向心 Catmull-Rom 曲线, `L` = 线性, `P` = 完美圆弧)
+- **`curvePoints` （用`|`分隔的字符串列表）：** 用于构造滑条的锚点。每个点的格式为 `x:y`.
+- **`slides` （整数）：** 玩家必须在滑条完成之前来回跟随滑条曲线的次数。也可以将其解释为往返计数加1。
+- **`length` （小数）：** 滑条的可见长度（单位为[osu! 像素](/wiki/Glossary#osupixel)）
+- **`edgeSounds` （用`|`分隔的整数列表）：**敲击滑条边时发出的音效。第一个声音是第一次单击滑块时播放的声音，最后一个声音是到达滑块末端播放的声音。
+- **`edgeSets` （用`|`分隔的字符串列表）：** 用于`edgeSounds`的采样集。每个采样集的格式为`normalSet:additionSet`，与 [打击音效部分](#Hitsounds（打击音效）)是一个意思.
+
+#### Slider curves（滑条曲线类型）
+
+当为滑条构建曲线时，`x`和`y`用于第一个控制点，剩下的为`curvePoints`。
+
+osu!有四种滑条曲线类型:
+
+- **贝兹曲线 （B）：** 可以任意使用[贝兹曲线](https://baike.baidu.com/item/%E8%B4%9D%E5%A1%9E%E5%B0%94%E6%9B%B2%E7%BA%BF) 。通过重叠它们的交点，可以将多个贝塞尔曲线合并到单个滑块中。
```

Comment:
I know wikipedia is blocked in China, but only changing the link here seems weird

Reply:
because Centripetal catmull-rom has no suitable document, so still use wikipedia

Reply:
```suggestion
- **贝兹曲线 （B）：** 可以任意使用[贝兹曲线](https://zh.wikipedia.org/wiki/貝茲曲線) 。通过重叠它们的交点，可以将多个贝塞尔曲线合并到单个滑块中。
```